scripts/quests/otherAreas/Test_My_Mettle.lua

```diff
@@ -0,0 +1,166 @@
+        player:delQuest(xi.quest.log_id.OTHER_AREAS, xi.quest.id.otherAreas.TEST_MY_METTLE)
+    end
+end
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            -- Quest attempted previously, must zone and wait until midnight
+            if
+                quest:getMustZone(player) or
+                vars.Wait >= os.time()
+            then
+                return false
+            end
+            -- Repeatable
+            if status == QUEST_COMPLETED then
+                return true
+            end
+            -- Quest requirements
+            return status == QUEST_AVAILABLE and
+                player:getMainLvl() >= 10 and
+                player:getFameLevel(xi.quest.fame_area.SELBINA_RABAO) >= 1
+        end,
+
+        [xi.zone.SELBINA] =
+        {
+            ['Devean'] = quest:progressEvent(120, xi.items.POWER_SANDALS, xi.items.POWER_SANDALS, 0, 0, xi.items.FLINT_STONE, xi.items.FLINT_STONE),
+
+            onEventFinish =
+            {
+                [120] = function(player, csid, option, npc)
+                    -- Option returns the sum of the gil option selected
+                    -- and the hour option selected, anything other than
+                    -- these combinations is invalid
+                    local possibleOptions =
+                    {
+                        124, 136, 148, 160, 172,
+                        274, 286, 298, 310, 322,
+                        524, 536, 548, 560, 572,
+                        1024, 1036, 1048, 1060, 1072,
+                    }
+                    local found = false
+                    for _, possibleOption in ipairs(possibleOptions) do
+                        if option == possibleOption then
+                            found = true
+                            break
+                        end
+                    end
+                    -- Rejected the quest at some point in the dialogue
+                    if found == false then
+                        return
+                    end
+
+                    -- Determine gil and hour options selected
+                    local gilOptions = { [1] = 100, [2] = 250, [5] = 500, [10] = 1000 }
+                    local gil = gilOptions[math.floor(option / 100)]
+                    local hours = option - gil
+
+                    -- Calculate potential payout
+                    local payoutRatios = { [24] = 3, [36] = 2.5, [48] = 2, [60] = 1.5, [72] = 1.2 }
+                    local gilPayout = payoutRatios[hours] * gil
+
+                    quest:setVar(player, 'GilPayout', gilPayout)
+                    quest:setVar(player, 'HoursSelected', hours)
+                    quest:setVar(player, 'HourStarted', VanadielHour())
+                    quest:setVar(player, 'DayStarted', VanadielDayOfTheYear())
+                    -- Remove quest from player's log
+                    player:delQuest(xi.quest.log_id.OTHER_AREAS, xi.quest.id.otherAreas.TEST_MY_METTLE)
```

Comment:
```suggestion
                    if status == QUEST_COMPLETED
                        player:delQuest(xi.quest.log_id.OTHER_AREAS, xi.quest.id.otherAreas.TEST_MY_METTLE)
                    end
```

Reply:
On retail the quest deletes itself from the log at this point, not sure if we should follow that

Reply:
Yeah no problem deleting the quest. but we can't delete it if it's not yet accepted. So we should just check quickly if the player has already completed it before deleting.